Code/Tools/LuaIDE/Source/LUA/LUAEditorMainWindow.cpp

```diff
@@ -818,19 +821,35 @@ namespace LUAEditor
 
     void LUAEditorMainWindow::OnFileMenuOpen()
     {
-        const QDir rootDir { AZ::Utils::GetProjectPath().c_str() };
-        const QString name = QFileDialog::getOpenFileName(this,
-                                                          "Open lua file",
-                                                          m_lastOpenFilePath.empty() ? rootDir.absolutePath() : m_lastOpenFilePath.c_str(),
-                                                          "Lua files (*.lua)");
-        if (name.isEmpty())
+        AssetSelectionModel selection;
+
+        StringFilter* stringFilter = new StringFilter();
+        stringFilter->SetName("Lua file (*.lua)");
+        stringFilter->SetFilterString(".lua");
+        stringFilter->SetFilterPropagation(AssetBrowserEntryFilter::PropagateDirection::Down);
+        auto stringFilterPtr = FilterConstType(stringFilter);
+
+        selection.SetDisplayFilter(stringFilterPtr);
+        selection.SetSelectionFilter(stringFilterPtr);
+
+        AssetBrowserComponentRequestBus::Broadcast(
+            &AssetBrowserComponentRequests::PickAssets, selection, AzToolsFramework::GetActiveWindow());
+
+        if (!selection.IsValid())
         {
             return;
         }
 
-        const AZStd::string assetId(name.toUtf8().data());
-        Context_DocumentManagement::Bus::Broadcast(&Context_DocumentManagement::Bus::Events::OnLoadDocument, assetId, true);
+        auto* result = selection.GetResult();
+
+        if (result == nullptr)
```

Comment:
I would do !result as more coherent with the rest of the code

Reply:
Right, thanks!